### PR TITLE
[PORT]: Nightvision Rework (In the name of color) from TG & full refactor of our render plane, layers, and rendering plate. Also ports their Camera + updates TGUI for it. And makes emissives work better and adds a bunch of them and refractors the bar sign to be a machine to use emissives.

### PR DESCRIPTION
**Pull request by cowbot92**
FULL CREDIT GOES TO THE ORIGINAL PR JESUS CHRIST https://github.com/tgstation/tgstation/pull/73094
also ports https://github.com/tgstation/tgstation/pull/73106 and https://github.com/tgstation/tgstation/pull/50109

Special thanks to @Chubbygummibear @Moltijoe @saraaah

# Document the changes in your pull request

This PR will completely rework the way Nightvision works, and how people generally see in the dark. 

Please refer to the PR linked above for all the technical jargon- I won't be able to explain it properly.

Additionally this will add emissive starlight as a bi-product, and potentially allow us to better control our lighting system

Finally, to do all of this I had to port over TG's Plane Master, Layers, Render Plate and several other things.

# Why is this good for the game?

I believe it looks a lot better, right now our night vision is kinda just... removal of darkness, which is fair but I think we can go so much further by using the colours provided to us. 

Emissive starlight is cool, that's about that.

Plane master needed updating as with everything else.

#Sprites

image
image

# Testing

EDIT: sechud nightvision was removed

https://github.com/yogstation13/Yogstation/assets/75333826/c44409ab-1f44-44d8-9744-eb6ab1814248

# Changelog

:cl:  Cowbot92 & ChubbyGummibear & Moltijoe

wip: Ports Nightvision from TG
tweak: Ports Nightvision from TG 
experimental: This is experimental

/:cl:

### CODE REVIEW

File: code/__DEFINES/sight.dm
Comment (Moltijoe) [inline code/__DEFINES/sight.dm]: don't forget this

Comment (cowbot92) [reply code/__DEFINES/sight.dm]: done

**Comment by iamthesilly** [comment]
very cool

**Comment by cowbot92** [comment]
Currently waiting for https://github.com/tgstation/tgstation/issues/79954 to be fixed via byond upgrades (or downgrades?), don't merge before its fixed

**Comment by MajManatee** [comment]
Now draw her being atomized


(I really don't care)

**Comment by cowbot92** [comment]
editors note: still waiting on byond update to fix camera net

**Comment by cowbot92** [comment]
editors note part 2: waiting on TG or byond to update their camera stuff

**Comment by cowbot92** [comment]
Chubby fixed the cameras how is he so good

**Review by neeshacark — APPROVED** [review]

**Review by cuackles — APPROVED** [review]

**Comment by PibeCantina** [comment]
Had a bug last round where me (ethereal) didnt produce light anymore after coming from lavaland, idk if this makes it happen but i tought i should let you know?
imagen_2024-01-06_050326531
Me when no light in mining

**Comment by cowbot92** [comment]
> Had a bug last round where me (ethereal) didnt produce light anymore after coming from lavaland, idk if this makes it happen but i tought i should let you know? imagen_2024-01-06_050326531 Me when no light in mining

do your other sources of light work?

**Comment by PibeCantina** [comment]
> > Had a bug last round where me (ethereal) didnt produce light anymore after coming from lavaland, idk if this makes it happen but i tought i should let you know? imagen_2024-01-06_050326531 Me when no light in mining
> 
> do your other sources of light work?

Yeah, i had my pka with a seclite and it would work normally when i took it out of my bag

**Comment by cowbot92** [comment]
things to test/fix before full merge:
Auto-observe parallax breaking
replays (lol)
check for stupid ethereal light breaking

**Comment by MajManatee** [comment]
It's simple, remove ethereals glow

**Comment by cowbot92** [comment]
> things to test/fix before full merge: Auto-observe parallax breaking replays (lol) check for stupid ethereal light breaking

the auto-observe issue is a tg/byond issue, so yeah.
replays will be worked on when chubby finishes https://github.com/yogstation13/Yogstation/pull/21221 (Im going to keep this one open until he's finished)
ethereal lights actually do work on his PR, so once I know it's all good ill close this and we'll merge his, which includes all my files already

**Comment by cowbot92** [comment]
chubby fixed all the problems & this is included in https://github.com/yogstation13/Yogstation/pull/21221 closing it

**Pull request closed without merging**
